backend/src/main/java/com/example/backend/Controllers/AdminDashboard/FlightManagementController.java

```diff
@@ -23,4 +26,16 @@ public ResponseEntity<String> deleteFlight(@PathVariable int id) {
         flightManagmentService.deleteFlight(id);
         return ResponseEntity.ok("The flight has been deleted successfully.");
     }
+
+    @DeleteMapping("/flights/cancel/{id}")
+    public ResponseEntity<String> cancelFlight(@PathVariable int id) {
+        flightManagmentService.cancelFlight(id);
+        return ResponseEntity.ok("The flight has been cancelled successfully.");
+    }
```

Comment:
Using @DeleteMapping in this context is not ideal because the flight isn't actually being removed; instead, its status is being updated. It would be more appropriate to use @PutMapping to reflect the action of modifying the flight's status. This approach aligns better with RESTful API principles and improves clarity in your endpoint's purpose.